File: C++/2D Array/Code2.cpp
//2D Array Print
#include<iostream>
using namespace std;

int main(){
    int matrix[4][3] = {{1, 2, 3},{4, 5, 6},{7, 8, 9},{10, 11, 12}};
    int rows = 4;
    int cols = 3;

    for(int i = 0; i <= rows - 1; i++){
        for(int j = 0; j <= cols - 1; j++){
            cout << matrix[i][j] << " ";
        }
        cout << endl;
    }
    cout << endl;
    return 0;
}

File: Java/2D Array/Code2.java
import java.util.Scanner;

public class Code2{
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter the number of Rows = ");
        int n = sc.nextInt();

        System.out.print("Enter the number of Columns = ");
        int m = sc.nextInt();

        int matrix[][] = new int[n][m];

        for(int i = 0; i <= matrix.length - 1; i++){// you could have written i < 3;
            for(int j = 0; j <= matrix[i].length - 1; j++){// you could have written j < 3;
                System.out.print("Enter your data for this cell = ");
                matrix[i][j] = sc.nextInt();
                //So, for a 2D array, matrix.length gives the number of rows 
                //(outer array length), 
                //while matrix[i].length gives the number of columns for 
                //the i-th row (inner array length)
            }
        }
        // Print the entire matrix
        for(int i = 0; i <= matrix.length - 1; i++){
            for(int j = 0; j <= matrix[i].length - 1; j++){
                System.out.print(matrix[i][j] + " ");
            }
            System.out.println();
        }

        sc.close();
    }
}
